utils/cache_manager.py

```
@@ -1,20 +1,131 @@
+    def set(self, key, value, timeout=120):
+        """Set data in the cache with a checksum for integrity."""
+        try:
+            if isinstance(value, Response):
+                value = {
+                    "response_data": value.get_data(as_text=True),
+                    "status": value.status_code,
+                    "headers": dict(value.headers),
+                }
+
+            # Serialize value and generate checksum
+            serialized_value = json.dumps(value, cls=jencoder)  # Serialize with jencoder
```

Ruff (0.8.0)

Local variable `serialized_value` is assigned to but never used

Remove assignment to unused variable `serialized_value`

(F841)